src/bin/cog_commit.rs

```diff
@@ -1,18 +1,44 @@
 use std::fmt::Write;
 
-use cocogitto::COMMITS_METADATA;
-
 use anyhow::{bail, Result};
-use conventional_commit_parser::commit::Separator;
+use conventional_commit_parser::commit::{CommitType, Separator};
 use itertools::Itertools;
 
+use cocogitto::COMMITS_METADATA;
+
 pub fn commit_types() -> Vec<&'static str> {
     COMMITS_METADATA
         .iter()
         .map(|(commit_type, _)| commit_type.as_ref())
         .collect()
 }
 
+pub fn explain_commit_type(commit_type: &str) -> String {
+    let learn_more_msg = "Learn more about supported commit types :\r
+- The Conventional Commit Specification : https://www.conventionalcommits.org \r
+- The Angular Commit Convention : https://github.com/angular/angular/blob/master/CONTRIBUTING.md#type \r
+- Custom commit types : https://docs.cocogitto.io/guide/#custom-commit-types";
+
+    let semantic_versioning_minor_msg = "Semantic versioning : this correlates with MINOR";
+    let semantic_versioning_patch_msg = "Semantic versioning : this correlates with PATCH";
+    let semantic_versioning_no_correlation_msg = "Semantic versioning : no correlation";
+
+    match commit_type {
```

Comment:
I'd suggest converting to `CommitType` with

```rust
let commit_type = CommitType::from(commit_type);
```

and then matching against that

Reply:
Good idea!